spec/consensus/light-client.md

```diff
@@ -1,318 +1,577 @@
-  follows the protocol until time _t_ (we will see about recovery later).
+    // returns true if commit corresponds to the block data in the header; otherwise false
+    func matchingCommit(header Header, commit Commit) bool
 
-### Tendermint Failure Model
+    // returns the set of validators from the given validator set that committed the block
+    // it does not assume signature verification
+    func signers(commit Commit, validatorSet ValidatorSet) []Validator
 
-If a block _h_ is generated at time _bfttime_ (and this time is stored in the block), then a set of validators that hold more than 2/3 of the voting power in h.Header.NextV is correct until time h.Header.bfttime + tp.
+    // return the voting power the validators in v1 have according to their voting power in set v2
+    // it assumes signature verification so it can be computationally expensive
+    func votingPowerIn(v1 []Validator, v2 ValidatorSet) int64
 
-Formally,
-\[
-\sum*{(v,p) \in h.Header.NextV \wedge correct(v,h.Header.bfttime + tp)} p >
-2/3 \sum*{(v,p) \in h.Header.NextV} p
-\]
+    // add this state as trusted to the store
+    func add(store Store, trustedState TrustedState) error
 
-_Assumption_: "correct" is defined w.r.t. realtime (some Newtonian global notion of time, i.e., wall time), while _bfttime_ corresponds to the reading of the local clock of a validator (how this time is computed may change when the Tendermint consensus is modified). In this note, we assume that all clocks are synchronized to realtime. We can make this more precise eventually (incorporating clock drift, accuracy, precision, etc.). Right now, we consider this assumption sufficient, as clock synchronization (under NTP) is in the order of milliseconds and _tp_ is in the order of weeks.
+    // retrieve the trusted state at given height if it exists (error = nil)
+    // return an error if there are no trusted state for the given height
+    // if height = 0, return the latest trusted state
+    func get(store Store, height int64) (TrustedState, error)
+```
 
-_Remark_: This failure model might change to a hybrid version that takes heights into account in the future.
+### Failure Model
 
-The specification in this document considers an implementation of the lite client under this assumption. Issues like _counter-factual signing_ and _fork accountability_ and _evidence submission_ are mechanisms that justify this assumption by incentivizing validators to follow the protocol.
-If they don't, and we have more that 1/3 faults, safety may be violated. Our approach then is to _detect_ these cases (after the fact), and take suitable repair actions (automatic and social). This is discussed in an upcoming document on "Fork accountability". (These safety violations include the lite client wrongly trusting a header, a fork in the blockchain, etc.)
+The lite client specification is defined with respect to the following failure model: If a block `b` is generated
```

Comment:
Maybe we can be more precise like, 

```
If a block `b` with header `h` is generated at time `Time` (ie. h.Time == Time), then ...
```

Also we have not defined TRUSTED_PERIOD, it just shows up in the equation at the end. Should we introduce it first ?